vault/resource_gcp_secret_backend.go

```diff
@@ -131,7 +131,7 @@ func gcpSecretBackendCreate(ctx context.Context, d *schema.ResourceData, meta in
 
 	d.Partial(true)
 	log.Printf("[DEBUG] Mounting GCP backend at %q", path)
-	useAPIVer117Ent := provider.IsAPISupported(meta, provider.VaultVersion117Ent)
+	useAPIVer117Ent := provider.IsAPISupported(meta, provider.VaultVersion117) && provider.IsEnterpriseSupported(meta)
```

Comment:
Was the previous check not sufficient to check for ent support?

Reply:
@fairclothjm upon testing with different versions I realized that since `IsAPISupported` only checks if a semantic version is greater than the other, the previous check was passing for non-enterprise versions since `v1.17.0` > `v1.16.3+ent`. It looks like the safest bet is to check the enterprise metadata separately from the version comparison

Reply:
Thanks that makes sense.  After this PR lands are we planning to update the other WIF TFVP resources to make the same check?

Reply:
Oh, looks like you may have covered those in this PR already with recent commits.